lib/model/folder_sendrecv.go

```diff
@@ -1880,8 +1909,35 @@ func (f *sendReceiveFolder) deleteDir(dir string, ignores *ignore.Matcher, scanC
 	return err
 }
 
+// itemChanged returns true if the given disk file differs from the information
+// in the database and schedules that file for scanning
+func (f *sendReceiveFolder) itemChanged(stat fs.FileInfo, item protocol.FileInfo, hasItem bool, scanChan chan<- string) (changed bool, err error) {
```

Comment:
I think a lot of stuff that is passed into this, could just be geneerated by this function itself, so you'd just need to pass in a name or protocol.FileInfo (to avoid the db access).

I wonder how often we'd end up with double stats because of this. If alot, perhaps not worth it.

Reply:
`itemChanged` is usually called when an item is there aka after stat returned a nil error. If stat returns an `IsNotExist` error, this is a special case and doesn't need the full rescan treatment, as a change will always win over a deletion in a conflict. To me it seems semantically simpler/clearer to first check and act on the stat result and if necessary, call `itemChanged`. I believe the correct thing would happen, if `itemChanged` returned false (file did not change) if there is no file on disk, but I wouldn't expect it to do that given it's name.